api/v1alpha1/nats_types.go

```diff
@@ -81,6 +63,70 @@ type NATS struct {
 	Status NATSStatus `json:"status,omitempty"`
 }
 
+// NATSStatus defines the observed state of NATS.
+type NATSStatus struct {
+	State      State              `json:"state"`
+	Conditions []metav1.Condition `json:"conditions,omitempty"`
+}
+
+// NATSSpec defines the desired state of NATS.
+type NATSSpec struct {
+	// Cluster defines configurations that are specific to NATS clusters.
+	Cluster Cluster `json:"cluster"`
+
+	// JetStream defines configurations that are specific to NATS JetStream.
+	// +optional
+	JetStream JetStream `json:"jetStream,omitempty"`
+
+	// JetStream defines configurations that are specific to NATS logging in NATS.
+	// +optional
+	Logging Logging `json:"logging,omitempty"`
+}
+
+// Cluster defines configurations that are specific to NATS clusters.
+type Cluster struct {
+	// Size of a NATS cluster, i.e. number of NATS nodes.
+	Size int `json:"size"`
+}
+
+// JetStream defines configurations that are specific to NATS JetStream.
+type JetStream struct {
+	// MemStorage defines configurations to memory storage in NATS JetStream.
+	// +optional
+	MemStorage MemStorage `json:"memStorage,omitempty"`
+
+	// FileStorage defines configurations to file storage in NATS JetStream.
+	// +optional
+	FileStorage FileStorage `json:"fileStorage,omitempty"`
+}
+
+// MemStorage defines configurations to memory storage in NATS JetStream.
+type MemStorage struct {
+	// Enable allows the enablement of memory storage.
+	Enable bool `json:"enable"`
+
+	// Size defines the mem.
+	Size string `json:"size"`
+}
+
+// FileStorage defines configurations to file storage in NATS JetStream.
+type FileStorage struct {
+	// StorageClassName defines the file storage class name.
+	StorageClassName string `json:"storageClassName"`
+
+	// Size defines the file storage size.
+	Size string `json:"size"`
+}
+
+// Logging defines logging options.
+type Logging struct {
+	// Debug allows debug logging.
+	Debug bool `json:"debug"`
+
+	// Trace allows trace logging.
+	Trace bool `json:"trace"`
+}
```

Comment:
Did you do `make build` to generate the CRD yaml file? and commit it?

Reply:
Now I have done so.